src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -83,7 +83,16 @@ CreateColocatedJoinChecker(Query *subquery, PlannerRestrictionContext *restricti
 		 * functions (i.e., FilterPlannerRestrictionForQuery()) rely on queries
 		 * not relations.
 		 */
+#if PG_VERSION_NUM >= PG_VERSION_16
+		RTEPermissionInfo *perminfo = NULL;
+		if (anchorRangeTblEntry->perminfoindex)
+		{
+			perminfo = getRTEPermissionInfo(subquery->rteperminfos, anchorRangeTblEntry);
+		}
+		anchorSubquery = WrapRteRelationIntoSubquery(anchorRangeTblEntry, NIL, perminfo);
```

Comment:
we might need WrapRteRelationIntoSubqueryCompat?

Reply:
I don't know how that would go.
However, I am thinking of defining RTEPermissionInfo prePG16 and passing it as NULL/NIL in these functions.

Reply:
sounds good